test/test_testing.py

```
@@ -1241,7 +1241,7 @@ def test_config_secret_option(self, request: pytest.FixtureRequest):
         assert harness.charm.changes == [{'name': 'config-changed', 'data': {'a': secret_id}}]
 
     def test_no_config_option_type(self):
-        with pytest.raises(RuntimeError):
+        with pytest.raises(KeyError):
```

We could explicitly raise a `RuntimeError` when the `'type'` key is missing to avoid this behaviour change?

---

I wondered about that. This should ever happen in production because charmcraft validates that there's a `type` when you pack, so you'd have to be deploying a non-packed charm or have messed with the installed `config.yaml`. It seemed like `RuntimeError` was the wrong kind of error for this, but on balance probably it is better to just be consistent.

---

This is definitely more annoying in the dataclass version, though...

---

Could wrap `self.config = {...` in a `try ... except` to catch the `KeyError` instead?

---

Or if the behaviour change is not a big deal, then just raise `KeyError`

---

> Could wrap `self.config = {...` in a `try ... except` to catch the `KeyError` instead?

Yes, although then the error can't say which option is faulty.

---

Ah, true, I retract that suggestion then.